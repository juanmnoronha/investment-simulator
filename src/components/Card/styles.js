import styled from 'styled-components';

export const Container = styled.section`
  align-items: center;
  background: #fff;
  border-radius: 8px;
  box-shadow: 0 0px 16px rgba(0, 0, 0, 0.1);
  display: flex;
  flex-direction: column;
  justify-content: center;
  max-width: 480px;
  padding: 36px;
  width: 100%;

  @media (max-width: 480px) {
    justify-content: flex-start;
    border-radius: 0;
    box-shadow: none;
    flex: 1;
    padding: 36px;
  }

  h1 {
    color: #324b68;
    font-size: 36px;
    font-weight: 500;
    margin-bottom: 16px;
    align-self: flex-start;

    @media (max-width: 480px) {
      margin-top: 16px;
    }
  }

  p {
    color: #333;
    font-size: 32px;
    margin-bottom: 36px;
    line-height: 48px;

    strong {
      color: #324b68;
      font-weight: 500;
    }
  }

  form {
    align-items: center;
    display: flex;
    flex-direction: column;
    flex: 1;
    width: 100%;

    input,
    select {
      background: transparent;
      border: 2px solid #324b68;
      border-radius: 4px;
      color: #324b68;
      font-size: 24px;
      font-weight: 500;
      margin-bottom: 24px;
      padding: 24px;
      transition: ease 0.2s;
      width: 100%;
    }

    input::placeholder,
    select {
      color: #324b68;
    }

    select {
      -webkit-appearance: none;
      background: transparent;
    }
  }
`;
