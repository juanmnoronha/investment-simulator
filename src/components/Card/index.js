import React from 'react';
import PropTypes from 'prop-types';
import { FaSpinner } from 'react-icons/fa';

import { Container } from './styles';

export default function Card({ children, icon, loading }) {
  return <Container>{loading ? icon : children}</Container>;
}

Card.defaultProps = {
  children: <div />,
  icon: <FaSpinner color="#6a4abe" size={48} />,
  loading: false,
};

Card.propTypes = {
  children: PropTypes.node,
  icon: PropTypes.node,
  loading: PropTypes.bool,
};
