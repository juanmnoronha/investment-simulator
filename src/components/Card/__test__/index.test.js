import React from 'react';
import { shallow } from 'enzyme';

import Card from '../index';

describe('<Card />', () => {
  it('should render properly', () => {
    const component = shallow(<Card />);
    expect(component).toMatchSnapshot();
  });
});
