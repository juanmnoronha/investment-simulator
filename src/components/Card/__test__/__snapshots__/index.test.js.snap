// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render properly 1`] = `ShallowWrapper {}`;
